paasta_tools/generate_deployments_for_service.py

```diff
@@ -104,6 +104,31 @@ def get_instance_config_for_service(soa_dir, service):
             )
 
 
+def get_latest_deployment_tag(refs, deploy_group):
+    """Gets the latest deployment tag and sha for the specified deploy_group
+
+    :param refs: A dictionary mapping git refs to shas
+    :param deploy_group: The deployment group to return a deploy tag for
+
+    :returns: A tuple of the form (ref, sha) where ref is the actual deployment
+              tag (with the most recent timestamp)  and sha is the sha it points at
+    """
+    most_recent_dtime = None
+    most_recent_ref = None
+    most_recent_sha = None
+    pattern = re.compile("^refs/tags/paasta-%s-(\d{8}T\d{6})-deploy$" % deploy_group)
```

Comment:
Do we? We have some similar functions that are used to create the tags (that generate the timestamp and support various desired_states). But I don't remember (and am not finding with grep) any other places with this particular regex. If you can find one, I would be happy to remove the duplication and create a new function that takes in a deploy_group as an argument and returns the compiled regex.